File: files/select_sort.py
def echanger(t, i, j):
    tmp = t[i]
    t[i] = t[j]
    t[j] = tmp


def index_du_plus_petit(t, k):
    n = len(t)
    imin = k
    for i in range(k + 1, n):
        if t[i] < t[imin]:
            imin = i
    return imin


def tri_selection(t):
    n = len(t)
    for k in range(n):
        imin = index_du_plus_petit(t, k)
        echanger(t, k, imin)


# TEST
L = [9, 2, 4, 5, 1, 7, 6, 8]
print(L)
tri_selection(L)
print(L)

File: files/select_sort.ml
let echanger t i j =
  let tmp = t.(i) in
  t.(i) <- t.(j);
  t.(j) <- tmp

let index_du_plus_petit t k =
  let n = Array.length t in
  let imin = ref k in
  for i = k + 1 to n - 1 do
    if t.(i) < t.(!imin) then imin := i (* L'affectation est bien une instruction () unit *)
  done;
  !imin

let tri_selection t =
  let n = Array.length t in
  for k = 0 to n - 1 do
    let imin = index_du_plus_petit t k in
    echanger t k imin
  done

(* TESTS *)
let a = [| 9; 0; 4; 5; 1; 3; 2; 7 |];;

a;;
tri_selection a;;
a
